packages/components/src/select-control/styles/select-control-styles.ts

```diff
@@ -57,7 +57,7 @@ const sizeStyles = ( { selectSize }: SelectProps ) => {
 		},
 	};
 
-	const style = sizes[ selectSize ] || sizes.default;
+	const style = sizes[ selectSize as Size ] || sizes.default;
```

Comment:
Instead of typecasting, maybe we could use a ternary?

```suggestion
	const style = selectSize ? sizes[ selectSize ] : sizes.default;
```

Reply:
Same as https://github.com/WordPress/gutenberg/pull/33714#discussion_r685872996